internal/test/output/output.go

```diff
@@ -24,22 +25,78 @@ func Assert(t *testing.T, actual string, expectedLines map[int]func(string) erro
 	}
 }
 
-// Prefix returns whether if the line has the specified string as prefix
-func Prefix(expected string) func(string) error {
-	return func(actual string) error {
+// AssertContains checks whether the output contains specified lines in order
+func AssertContains(t *testing.T, actual string, expectedLines []func(string, bool) error) {
```

Comment:
Why is this necessary? Previously it was working with `Assert`.

Reply:
I added this function so that the build test would pass against Docker EE which uses a 17.06 engine. With this engine, `COPY` steps have an extra `Removing intermediate container` line. This new function doesn't check exact line numbers but instead checks for desired outputs in order.

**17.06:**
```bash
$ docker build -t copy .
Sending build context to Docker daemon   2.56kB
Step 1/2 : FROM alpine
 ---> 3fd9065eaf02
Step 2/2 : COPY something /
 ---> 577b61f8471b
Removing intermediate container 86ec5fc54328
Successfully built 577b61f8471b
Successfully tagged copy:latest
```

**18.06:**
```bash
$ docker build -t copy .
Sending build context to Docker daemon   2.56kB
Step 1/2 : FROM alpine
 ---> 3fd9065eaf02
Step 2/2 : COPY something /
 ---> 93321d321367
Successfully built 93321d321367
Successfully tagged copy:latest
```

Reply:
> Removing intermediate container 86ec5fc54328

This line disappears ? is it by design ? cc @thaJeztah @simonferquel

Reply:
Was wondering the same when reading that; I don't think that line should disappear, will have to check why it's not printed.

Can confirm it's not there though;

```bash
mkdir foo && cd foo
echo bla > something

docker build --no-cache . -f-<<EOF
FROM alpine
COPY something /
COPY something /else/
EOF
```

```
Sending build context to Docker daemon  2.607kB
Step 1/3 : FROM alpine
 ---> 3fd9065eaf02
Step 2/3 : COPY something /
 ---> Using cache
 ---> 97e7adea2403
Step 3/3 : COPY something /else/
 ---> b0cdf35d65d6
Successfully built b0cdf35d65d6
Successfully tagged copy:latest
```

Reply:
Same result when adding `--rm` and `--force-rm` (just in case);

```
docker build --no-cache --force-rm --rm . -f-<<EOF
FROM alpine
COPY something /
COPY something /else/
EOF
```

```
Sending build context to Docker daemon  2.607kB
Step 1/3 : FROM alpine
 ---> 3fd9065eaf02
Step 2/3 : COPY something /
 ---> cf910fa96aa7
Step 3/3 : COPY something /else/
 ---> 56bcd1f3113b
Successfully built 56bcd1f3113b
```

Reply:
Instead of making the assertion less strict (and less obvious) I would prefer to have different expected values based on the version.

Reply:
@thaJeztah Thanks for the investigation. Chatting to @simonferquel offline; he believes the missing line is a bug.

@dnephin My thinking was that we should just be looking for lines of interest in order and that was enough of a test. I agree with you that it makes the check less clear. Since this appears to be an actual bug; I will wait for it to be fixed and then remove the check function.

Reply:
Created a moby issue for this: https://github.com/moby/moby/issues/37164

Updated PR to remove the addition of `AssertContains`.